lib/SILOptimizer/Differentiation/TangentBuilder.cpp

```diff
@@ -0,0 +1,202 @@
+//===--- TangentBuilder.cpp - Tangent SIL builder ------------*- C++ -*----===//
```

Comment:
Nit: A C++ marker is only needed  for .h files, where it's ambiguous whether it contains C or C++ code. A `.cpp` file always contains C++ code.